### DISCUSSION TITLE
Assign wildcard subdomain to deployment

### DISCUSSION THREAD

**Discussion by danmichaelo** [vercel/vercel#7273] [Help] [answered]
Hi! Trying to deploy a platform app to Vercel. 

Given a successful deploy `cakefactory-qa1apft8d-example.vercel.app` for a `chocolate` feature branch, how can we assign a subdomain like `chocolate.staging.example.org` and its subdomains `*.chocolate.staging.example.org` to that specific deploy? 

Here's what I'm currently trying: (I've replaced our domain with `example.org` in the examples below)

## ✅ 1. Create DNS records

This works:

```
$ vercel dns add example.org 'chocolate.staging' CNAME cname.vercel-dns.com
Vercel CLI 23.1.2
> Success! DNS record for domain example.org (rec_0e48d8f6334343432) created under example [708ms]

$ vercel dns add example.org '*.chocolate.staging' CNAME cname.vercel-dns.com
Vercel CLI 23.1.2
> Success! DNS record for domain example.org (rec_0e48d8f6334343432) created under example [708ms]
```

## ✅ 2. Create certificate

This works:

```
$ vercel certs issue 'chocolate.staging.example.org' '*.chocolate.staging.example.org'
```

## ✅ 3. Create alias for `chocolate.staging.example.org`

This works:

```
$ vercel alias set cakefactory-qa1apft8d-example.vercel.app 'chocolate.staging.example.org'
Vercel CLI 23.1.2
> Assigning alias chocolate.staging.example.org to deployment cakefactory-qa1apft8d-example.vercel.app
> Success! https://chocolate.staging.example.org now points to https://cakefactory-qa1apft8d-example.vercel.app [5s]
```

## 💣 4. Create alias for `*.chocolate.staging.example.org`

This fails:

```
$ vercel alias set cakefactory-qa1apft8d-example.vercel.app '*.chocolate.staging.example.org'
Vercel CLI 23.1.2
> Assigning alias *.chocolate.staging.example.org to deployment cakefactory-qa1apft8d-example.vercel.app
Error! We could not alias since the domain example.org could not be verified due to the following reasons:

Nameservers verification failed since we see a different set than the intended set:
     Intended Nameservers    Current Nameservers
     ns1.vercel-dns.com      ns1.vercel-dns.com     ✔
     ns2.vercel-dns.com      ns2.vercel-dns.com     ✔

  Read more: https://err.sh/vercel/domain-verification
```

*The last error message is not very helpful, since the intended and current nameservers are indeed the same.*

**Update**: I was able to create the alias using a CURL request instead, so perhaps this is a bug in Vercel CLI?

```
curl --location --request POST 'https://api.vercel.com/v3/deployments/dpl_example/aliases?teamId=team_example' \
--header 'Authorization: Bearer SECRET' \
--header 'Content-Type: application/json' \
--data-raw '{
    "alias": "*.chocolate.staging.example.org"
}'
```

**Comment by Amit0617** [comment]
You was supposed to use double quotes when using vercel Cli? Because terminal resolves wildcards only in double quotes.

**Reply by danmichaelo** [reply]
The argument to `--data-raw` was quoted using single quotes, so wildcard expansion in the shell was not the issue.

**Comment by michielglibert** [comment, marked as answer]
I also had this issue and was on the exact same version of the Vercel CLI. Updating it to the latest version resolved the issue for me.

**Reply by danmichaelo** [reply]
Same here, seems to work fine with the latest version 🎉
